### PR TITLE
Adds new method on URL called droppedScheme

### PR DESCRIPTION
**Pull request by sammy-SC**
Resolves: #464

## Checklist

- [ ] I checked the [**Contributing Guidelines**](https://github.com/SwifterSwift/SwifterSwift/blob/master/CONTRIBUTING.md) before creating this request.
- [x] New extensions are written in Swift 4.
- [x] New extensions support iOS 8.0+ / tvOS 9.0+ / macOS 10.10+ / watchOS 2.0+.
- [x] I have added tests for new extensions, and they passed.
- [x] All extensions have a **clear** comments explaining their functionality, all parameters and return type in English.
- [x] All extensions are declared as **public**.
- [ ] I have added a [changelog](https://github.com/SwifterSwift/SwifterSwift/blob/master/CHANGELOG_GUIDELINES.md) entry describing my changes.

I'm happy to make adjustments if required.

### CODE REVIEW

File: Tests/FoundationTests/URLExtensionsTests.swift [hunk 1]
```diff
@@ -71,4 +71,17 @@ final class URLExtensionsTests: XCTestCase {
 	}
 	#endif
 
+    func testDropScheme() {
+        let urls: [String: String] = [
+            "https://domain.com/path/other/": "domain.com/path/other/",
+            "https://domain.com": "domain.com",
+            "http://domain.com": "domain.com",
+            "file://domain.com/image.jpeg": "domain.com/image.jpeg"
```
Comment (guykogus) [inline Tests/FoundationTests/URLExtensionsTests.swift]: I suggest adding some of the urls listed in the original issue. E.g.
```
"://apple.com"
"//apple.com"
"apple.com"
```

Comment (guykogus) [reply Tests/FoundationTests/URLExtensionsTests.swift]: Also make sure to test `"http://"`.
And what do you think should be the outcome of `//`? Should it be `nil` or `//`? Currently it returns the latter, but to me it makes more sense for the former. Not an important edge case, but it's a curious one.

Comment (sammy-SC) [reply Tests/FoundationTests/URLExtensionsTests.swift]: @guykogus 

I've added test for `http://`.

I believe outcome of `//` should be `nil`, however I would need to add specific condition to detect this case, otherwise I'm unable to detect it. Shall I do that?

File: Sources/Extensions/Foundation/URLExtensions.swift [hunk 2]
```diff
@@ -102,6 +102,22 @@ public extension URL {
...
+        // https://github.com/SwifterSwift/SwifterSwift/issues/464
+        if let scheme = self.scheme {
+            return URL(string: String(self.absoluteString.dropFirst(scheme.count + 3)))
+        }
+
+        guard host != nil else {
+            return self
+        }
+
+        return URL(string: String(absoluteString.dropFirst(2)))!
```
Comment (LucianoPAlmeida) [inline Sources/Extensions/Foundation/URLExtensions.swift]: If the return is optional, why force unwrap here?

File: Sources/Extensions/Foundation/URLExtensions.swift [hunk 3]
```diff
@@ -102,6 +102,22 @@ public extension URL {
         }
     }
 
+    /// SwifterSwift: Generates new URL that does not have scheme.
+    ///
+    ///        let url = URL(string: "https://domain.com")!
+    ///        print(url.droppedScheme()) // prints "domain.com"
+    public func droppedScheme() -> URL? {
+        // https://github.com/SwifterSwift/SwifterSwift/issues/464
```
Comment (SD10) [inline Sources/Extensions/Foundation/URLExtensions.swift]: We don't need to put a reference to the issue here

File: Sources/Extensions/Foundation/URLExtensions.swift (hunk 2 above)
Comment (SD10) [reply Sources/Extensions/Foundation/URLExtensions.swift]: It would be nice to extract the inner expression

File: Tests/FoundationTests/URLExtensionsTests.swift (hunk 1 above)
Comment (guykogus) [reply Tests/FoundationTests/URLExtensionsTests.swift]: No, I don't think it's necessary.

File: Sources/Extensions/Foundation/URLExtensions.swift (hunk 3 above)
Comment (sammy-SC) [reply Sources/Extensions/Foundation/URLExtensions.swift]: @SD10 I've removed the reference.

File: Sources/Extensions/Foundation/URLExtensions.swift (hunk 2 above)
Comment (sammy-SC) [reply Sources/Extensions/Foundation/URLExtensions.swift]: @SD10 I've extracted the inner expression, in both cases where `URL(string:)` is used.

@LucianoPAlmeida removed force unwrap.

File: Tests/FoundationTests/URLExtensionsTests.swift [hunk 4]
```diff
@@ -71,4 +71,24 @@ final class URLExtensionsTests: XCTestCase {
...
+            "file://domain.com/image.jpeg": "domain.com/image.jpeg",
+            "://apple.com": "apple.com",
+            "//apple.com": "apple.com",
+            "apple.com": "apple.com",
+            "http://": nil,
+            "//": "//"
+        ]
+
+        urls.forEach { input, expected in
+            let url = URL(string: input)!
```
[the remark this answers is not in the document]
Comment (sammy-SC) [reply Tests/FoundationTests/URLExtensionsTests.swift]: @guykogus I've removed force-unwrap, from tests and from implementation.

**Comment by LucianoPAlmeida** [comment]
@sammy-SC Can you also fix the swiftlint warning pointed by danger :))

File: CHANGELOG.md [hunk 5]
```diff
@@ -8,6 +8,9 @@ The changelog for **SwifterSwift**. Also see the [releases](https://github.com/S
   - Added `addGestureRecognizers(_:)` which accepts an array of `UIGestureRecognizer` to add multiple gesture recognizers to a view with one call. [#523](https://github.com/SwifterSwift/SwifterSwift/pull/523) by [moyerr](https://github.com/moyerr)
   - Added `removeGestureRecognizers(_:)` which accepts an array of `UIGestureRecognizer` to remove multiple gesture recognizers from a view with one call. [#523](https://github.com/SwifterSwift/SwifterSwift/pull/523) by [moyerr](https://github.com/moyerr)
 
+- **URL**
+  - Added `droppedScheme()` which returns new URL that does not have scheme. [#528](https://github.com/SwifterSwift/SwifterSwift/pull/528) by [sammy-sc](https://github.com/sammy-SC)
```
Comment (marcocapano) [inline CHANGELOG.md]: Maybe we should use `URL` instead of URL?

Comment (marcocapano) [reply CHANGELOG.md]: I mean with backticks

File: Sources/Extensions/Foundation/URLExtensions.swift [hunk 6]
```diff
@@ -102,6 +102,23 @@ public extension URL {
...
+    ///
+    ///        let url = URL(string: "https://domain.com")!
+    ///        print(url.droppedScheme()) // prints "domain.com"
+    public func droppedScheme() -> URL? {
+        if let scheme = self.scheme {
+            let droppedScheme = String(self.absoluteString.dropFirst(scheme.count + 3))
+            return URL(string: droppedScheme)
+        }
+
+        guard host != nil else {
```
Comment (marcocapano) [inline Sources/Extensions/Foundation/URLExtensions.swift]: Would you mind moving the else block on the same line?

File: CHANGELOG.md (hunk 5 above)
Comment (sammy-SC) [reply CHANGELOG.md]: Done

File: Sources/Extensions/Foundation/URLExtensions.swift (hunk 6 above)
Comment (sammy-SC) [reply Sources/Extensions/Foundation/URLExtensions.swift]: Done

**Comment by sammy-SC** [comment]
@LucianoPAlmeida fixed.

Should I squash commit messages together?

**Comment by SD10** [comment]
Thank you for contributing to SwifterSwift! I've invited you to join the SwifterSwift GitHub organization - no pressure to accept! If you'd like more information on what that means, check out our [contributing guidelines](https://github.com/SwifterSwift/SwifterSwift/blob/master/CONTRIBUTING.md). Feel free to reach out if you have any questions! 😃
